### ISSUE TITLE
Can't Build using 2.0.1

### ISSUE THREAD

**Issue by jwogan5** [hansemannn/titanium-firebase-cloud-messaging#39]
I am using Appc SDK 7.5.0.   With Firebase core 2.0.0 and an older version of this module, everything works fine.  ( As you can see I am using the newest firebase core ).

• When I try to build to a device I get an error about "DTDeviceKit: deviceType from 00008006-000668380CE2002E was NULL"

• When I try to build to a simulator I get errors about "Undefined symbols for architecture x86_64"

Error log for build against simulator:

[TRACE]  Undefined symbols for architecture x86_64:
[TRACE]    "_OBJC_CLASS_$_FIRComponentType", referenced from:
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessaging.o)
[TRACE]    "_OBJC_CLASS_$_GULReachabilityChecker", referenced from:
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessaging.o)
[TRACE]    "_OBJC_CLASS_$_FIRComponent", referenced from:
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessaging.o)
[TRACE]    "_OBJC_CLASS_$_GULAppEnvironmentUtil", referenced from:
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessagingReceiver.o)
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessagingAnalytics.o)
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessagingUtilities.o)
[TRACE]    "_OBJC_CLASS_$_FIRDependency", referenced from:
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessaging.o)
[TRACE]    "_OBJC_CLASS_$_FIRComponentContainer", referenced from:
[TRACE]        objc-class-ref in FirebaseMessaging(FIRMessaging.o)
[TRACE]  ld: symbol(s) not found for architecture x86_64
[TRACE]  clang: error: linker command failed with exit code 1 (use -v to see invocation)
[ERROR]  ** BUILD FAILED **

Error log for build against device:

[TRACE]  CompileC /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.o /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/Classes/TiUIiOSMenuPopupProxy.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target: Dealer Center)
[TRACE]      cd /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone
[TRACE]      export LANG=en_US.US-ASCII
[TRACE]      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch arm64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=c99 -fmodules -gmodules -fmodules-cache-path=/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wunreachable-code -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wno-parentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wno-unused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -DDEPLOYTYPE=development -DLAUNCHSCREEN_STORYBOARD=1 -DDEFAULT_BGCOLOR_RED=1 -DDEFAULT_BGCOLOR_GREEN=1 -DDEFAULT_BGCOLOR_BLUE=1 -DTI_LOG_SERVER_PORT=45296 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -miphoneos-version-min=9.0 -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -Wno-unguarded-availability -index-store-path /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/DerivedData/Index/DataStore -iquote /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-generated-files.hmap -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-own-target-headers.hmap -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-all-target-headers.hmap -iquote /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-project-headers.hmap -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Products/Debug-iphoneos/include -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk/usr/include/libxml2 -Iheaders -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/DerivedSources/arm64 -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/DerivedSources -Wno-arc-performSelector-leaks -F/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Products/Debug-iphoneos -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/firebase.core/2.0.0/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/com.traderinteractive.appsee/0.0.1/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/com.traderinteractive.fyusion/0.0.2/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/firebase.cloudmessaging/2.0.1/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/firebase.analytics/1.3.0/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/ti.crashlytics/1.1.1/platform -DDEBUG -include /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/PrecompiledHeaders/SharedPrecompiledHeaders/11076202699044268272/Dealer_Center_Prefix.pch -MMD -MT dependencies -MF /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.d --serialize-diagnostics /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.dia -c /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/Classes/TiUIiOSMenuPopupProxy.m -o /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.o
[ERROR]  2018-11-26 15:54:12.250 xcodebuild[17594:1076757]  DTDeviceKit: deviceType from 00008006-000668380CE2002E was NULL

**Comment by m1ga** [comment]
same here with 2.0.1 + 7.4.1.GA

**Comment by jwogan5** [comment]
So I actually got the new versions of firebase analytics and firebase cloud messaging modules working today.  Basically, the latest release version of firebase core (2.0.0) is not the latest one that works with the new modules.  The new modules require firebase core ( 2.1.0 ).  I cloned the firebase core code and then built it myself which built the 2.1.0 module. I also added x86_64 and i386 to all of the arch versions within Xcode for all 3 modules. Once I built all 3 modules myself with the latest code from master my app built successfully to both the simulator and my device.

If I come across any other issues I will update this issue.
